### PR TITLE
Allow admin customization

### PR DESCRIPTION
**Pull request by soupette**
#### Description of what you did:
Enable front-end development directly from a generated project.

#### My PR is a:

- [ ] 💥 Breaking change
- [ ] 🐛 Bug fix
- [X] 💅 Enhancement
- [ ] 🚀 New feature

#### Main update on the:

- [X] Admin
- [ ] Documentation
- [ ] Framework
- [ ] Plugin

#### Manual testing done on the following databases:

- [ ] Not applicable
- [ ] MongoDB
- [ ] MySQL
- [ ] Postgres
- [ ] SQLite

### CODE REVIEW

File: docs/3.0.0-beta.x/advanced/customize-admin.md [hunk 1]
```diff
@@ -29,9 +29,101 @@ By default, the administration panel is exposed via [http://localhost:1337/admin
 
 The panel will be available through [http://localhost:1337/dashboard](http://localhost:1337/dashboard) with the configurations above.
 
+---
+
 ### Development mode
 
-**_Currently not available_**
+to enable the front-end development mode you need to start your application using the `--watch-admin` flag.
```
Comment (derrickmehaffy) [inline docs/3.0.0-beta.x/advanced/customize-admin.md]: `To enable`

File: docs/3.0.0-beta.x/advanced/customize-admin.md [hunk 2]
```diff
@@ -29,9 +29,101 @@ By default, the administration panel is exposed via [http://localhost:1337/admin
...
+
+```bash
+cd my-app
+strapi develop --watch-admin
+```
+
+With this option you can
```
Comment (derrickmehaffy) [inline docs/3.0.0-beta.x/advanced/customize-admin.md]: `you can do the following:`

File: docs/3.0.0-beta.x/advanced/customize-admin.md [hunk 3]
```diff
@@ -29,9 +29,101 @@ By default, the administration panel is exposed via [http://localhost:1337/admin
...
+
+#### Customizing the `strapi-admin` package
```
Comment (derrickmehaffy) [inline docs/3.0.0-beta.x/advanced/customize-admin.md]: `Customize`

File: docs/3.0.0-beta.x/advanced/customize-admin.md [hunk 4]
```diff
@@ -33,9 +33,101 @@ By default, the administration panel is exposed via [http://localhost:1337/admin
...
+
+    return obj;
+  },
+  {}
+);
+
+export { languages, translationMessages };
+```
+
+> With this modification only English and French will be available in your admin
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/advanced/customize-admin.md]: Can you please replace the `>`
by the following structure.
```
::: note
With this...
:::
```

File: docs/3.0.0-beta.x/advanced/customize-admin.md [hunk 5]
```diff
@@ -33,9 +33,101 @@ By default, the administration panel is exposed via [http://localhost:1337/admin
...
+**Example: Changing the available locales of your application**
+
+```bash
+# Create both the admin and admin/src folders
+cd my-app && mkdir -p admin/src
+# Change the available locales of the administration panel
+touch admin/src/i18n.js
+```
+
+**Path: `my-app/admin/src/i18n.js**
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/advanced/customize-admin.md]: Here is a missing **`** at the end of the line.

File: docs/3.0.0-beta.x/advanced/customize-admin.md [hunk 6]
```diff
@@ -33,9 +33,101 @@ By default, the administration panel is exposed via [http://localhost:1337/admin
...
+mkdir content-manager && cd content-manager
+# Create the admin folder
+mkdir -p admin/src
+# Create the components folder and the WysiwygWithErrors one
+cd admin/src && mkdir -p components/WysiwygWithErrors
+# Create the index.js so the original file is overridden
+touch components/WysiwygWithErrors/index.js
+```
+
+**Path: `my-app/extensions/content-manager/admin/src/components/WysiwygWithErrors/index.js**
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/advanced/customize-admin.md]: Here is a missing **`** at the end of the line.

File: docs/3.0.0-beta.x/advanced/customize-admin.md (hunk 5 above)
Comment (lauriejim) [reply docs/3.0.0-beta.x/advanced/customize-admin.md]: And can you follow this format https://github.com/strapi/strapi/pull/4152/files#diff-67f3303196b46a0cd7c1d31b87dfca5dR19

File: docs/3.0.0-beta.x/advanced/customize-admin.md (hunk 6 above)
Comment (lauriejim) [reply docs/3.0.0-beta.x/advanced/customize-admin.md]: And can you follow this format please https://github.com/strapi/strapi/pull/4152/files#diff-67f3303196b46a0cd7c1d31b87dfca5dR19

File: docs/3.0.0-beta.x/advanced/customize-admin.md [hunk 7]
```diff
@@ -33,9 +33,101 @@ By default, the administration panel is exposed via [http://localhost:1337/admin
...
+```
+
+**Path: `my-app/admin/src/i18n.js**
+
+```js
+import { addLocaleData } from 'react-intl';
+import { reduce } from 'lodash';
+import en from 'react-intl/locale-data/en';
+import fr from 'react-intl/locale-data/fr';
+
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/advanced/customize-admin.md]: If I **copy/paste** this snippet code. I have an issue on `trads` variable.
I checked the origin the `import trads from './translations';`is missing.

File: docs/3.0.0-beta.x/advanced/customize-admin.md [hunk 8]
```diff
@@ -33,9 +33,101 @@ By default, the administration panel is exposed via [http://localhost:1337/admin
...
+strapi develop --watch-admin
+```
+
+With this option you can do the following:
+
+#### Customize the `strapi-admin` package
+
+All files added in `my-app/admin/src/` will either be replaced or added
+
+**Example: Changing the available locales of your application**
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/advanced/customize-admin.md]: For this change, if I just update the following file, all translations still there.

**Comment by soupette** [comment]
> Thank. you @soupette and @alexandrebodin for this PR. I made some comments.
> 
> And I would like to know more about this [documentation page](http://localhost:8081/documentation/3.0.0-beta.x/plugin-development/frontend-development.html).
> Is there an update to do with this PR.

Well this PR enables the development in the administration panel. In order to develop custom plugins we need to update the plugin generators with the front-end files. Furthermore, before writing any more documentation I would like to complete the migration to `styled-components` and remove `redux-sagas` first. Plus we need to add [Buffetjs](https://buffetjs.io) in the admin which is our next task.

So to answer your question I believe that it best to leave this doc as it is, and update it once we have achieved all the task specified above.

File: docs/3.0.0-beta.x/advanced/customize-admin.md (hunk 7 above)
Comment (soupette) [reply docs/3.0.0-beta.x/advanced/customize-admin.md]: I have updated the example.

File: docs/3.0.0-beta.x/advanced/customize-admin.md (hunk 8 above)
Comment (soupette) [reply docs/3.0.0-beta.x/advanced/customize-admin.md]: Same

**Comment by kayue** [comment]
Is this feature part of the roadmap? Want to know when can we develop custom backend in Strapi.

**Comment by derrickmehaffy** [comment]
@kayue based on @soupette 's reply no it's not on the roadmap yet, so there is no ETA for custom plugin AdminUI integration yet.

**Comment by lauriejim** [comment]
You did, but I have to re-test (sorry for the confusing message)

**Pull request merged**